### ISSUE TITLE
Xenforo title parsing fails with nested brackets

### ISSUE THREAD

**Issue by hseg** [JimmXinu/FanFicFare#782]
Examples: https://forums.spacebattles.com/threads/581719, https://forums.spacebattles.com/threads/506906
In both, the title ends in `[Worm (Alt!Power)/X]`. causing the tag-extracting logic to believe the tag is `[Worm (Alt!Power)` and leaving the `/X]` in the title.
Presumably, this is because the parser stops at any closing bracket, not a matching closing bracket.

Produced on `fanficfare 4.8.5 (61bc7328)`

**Comment by JimmXinu** [comment]
The title normalization is in [defaults.ini#L735](https://github.com/JimmXinu/FanFicFare/blob/main/fanficfare/defaults.ini#L735), and the `tagsfromtitle` patterns are above it.  (And in [plugin-defaults.ini](https://github.com/JimmXinu/FanFicFare/blob/main/calibre-plugin/plugin-defaults.ini) of course.)

If you have a regex that can handle that more correctly, I'd be happy to see it.  

I am resigned to it being imperfect as long as authors include 'tags' in the title.  My own personal.ini, for example, contains:
```
# Powers that (Shouldn't) Be
[https://forums.spacebattles.com/threads/567184/]
add_to_replace_metadata:
 title=>^.*$=>Powers that (Shouldn't) Be
 tagsfromtitle=>Shouldn't=>Worm
```

**Comment by hseg** [comment]
OK, looking at the regexes, just need to duplicate the logic for `[]` and `()`.
Writing this now.

**Comment by hseg** [comment]
In testing this, noticed (especially visible with https://forums.spacebattles.com/threads/631468) that the tags aren't stripped for spaces, and are reversed from how they appear in-title.

**Comment by JimmXinu** [comment]
Default metadata list handling orders items alphabetically.  Set `keep_in_order_tagsfromtitle:true` if you want them in the original order.

The extra space doesn't matter 99% of the time, but should be easily addressed by adding comma to the search pattern in [defaults.ini#L723](https://github.com/JimmXinu/FanFicFare/blob/main/fanficfare/defaults.ini#L723)

**Comment by JimmXinu** [comment]
Addressed in #783.
